ui/src/app/machines/views/MachineList/NameColumn/NameColumn.js

```diff
@@ -0,0 +1,101 @@
+import { useSelector } from "react-redux";
+import React from "react";
+import PropTypes from "prop-types";
+
+import { machine as machineSelectors } from "app/base/selectors";
+import Tooltip from "app/base/components/Tooltip";
+
+const generateFQDN = (machine, machineURL) => {
+  const name = (
+    <a href={machineURL} title={machine.fqdn}>
+      <strong>
+        {machine.locked ? (
+          <span title="This machine is locked. You have to unlock it to perform any actions.">
+            <i className="p-icon--locked">Locked: </i>{" "}
+          </span>
+        ) : null}
+        {machine.hostname}
+      </strong>
+      <small>.{machine.domain.name}</small>
+    </a>
+  );
+  let ipAddresses = [];
+  let bootIP;
+  machine.ip_addresses.forEach(address => {
```

Comment:
This needs a guard, as machines are not guaranteed to have `ip_addresses`. The listing crashes otherwise.

Reply:
Done, mind taking another look?